lib/active_merchant/billing/gateways/iridium.rb

```diff
@@ -291,7 +291,8 @@ def build_reference_request(type, money, authorization, options)
         order_id, cross_reference, _ = authorization.split(';')
         build_request(options) do |xml|
           if money
-            details = {'CurrencyCode' => currency_code(options[:currency] || default_currency), 'Amount' => amount(money)}
+            currency = options[:currency] || currency(money)
+            details = {'CurrencyCode' => currency_code(options[:currency] || default_currency), 'Amount' => localized_amount(money, currency)}
```

Comment:
Would it make sense to use the `currency` variable in the currency_code method?

Reply:
Thanks @molbrown, that does make more sense. This has been updated.

Reply:
👍